src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java

```diff
@@ -1,223 +1,22 @@
 // Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 
 package org.intellij.sdk.codesync;
-import kotlin.Pair;
-
-import com.intellij.ide.startup.StartupManagerEx;
-import com.intellij.openapi.application.ApplicationManager;
-import com.intellij.openapi.editor.EditorFactory;
-import com.intellij.openapi.editor.event.DocumentEvent;
-import com.intellij.openapi.editor.event.DocumentListener;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.project.ProjectManagerListener;
-import com.intellij.openapi.roots.ProjectFileIndex;
-import com.intellij.openapi.roots.ProjectRootManager;
-import com.intellij.openapi.util.Disposer;
-import com.intellij.openapi.vfs.VirtualFile;
-import com.intellij.openapi.vfs.VirtualFileManager;
-import com.intellij.openapi.vfs.newvfs.BulkFileListener;
-import com.intellij.openapi.vfs.newvfs.events.VFileEvent;
 import com.intellij.serviceContainer.AlreadyDisposedException;
-import kt.org.intellij.sdk.codesync.tasks.TaskExecutor;
-import org.apache.commons.codec.digest.DigestUtils;
-import org.intellij.sdk.codesync.alerts.ActivityAlerts;
-import org.intellij.sdk.codesync.codeSyncSetup.CodeSyncSetup;
 import org.intellij.sdk.codesync.database.SQLiteConnection;
-import org.intellij.sdk.codesync.database.migrations.MigrateRepo;
-import org.intellij.sdk.codesync.database.migrations.MigrationManager;
-import org.intellij.sdk.codesync.exceptions.common.FileNotInModuleError;
 import org.intellij.sdk.codesync.locks.CodeSyncLock;
-import org.intellij.sdk.codesync.state.RepoStatus;
-import org.intellij.sdk.codesync.state.StateUtils;
 import org.intellij.sdk.codesync.utils.CommonUtils;
-import org.intellij.sdk.codesync.utils.FileUtils;
-import org.intellij.sdk.codesync.utils.ProjectUtils;
 import org.jetbrains.annotations.NotNull;
 
-import java.io.IOException;
 import java.sql.SQLException;
-import java.time.Instant;
-import java.time.temporal.ChronoUnit;
-import java.util.*;
 
 import static org.intellij.sdk.codesync.Constants.*;
-import static org.intellij.sdk.codesync.Utils.*;
-import static org.intellij.sdk.codesync.codeSyncSetup.CodeSyncSetup.createSystemDirectories;
 
 /**
  * Listener to detect project open and close.
  */
 public class ProjectOpenCloseListener implements ProjectManagerListener {
```

Comment:
I see we are not using `ProjectOpenCloseListener` listner anymore, instead we are now using `postStartupActivity`, what's the difference in behaviour. 
Is `postStartupActivity` not dependent on the opened project and starts the Plugin as soon IDE is opened?

Reply:
`ProjectOpenCloseListener ` is deprecated and its usage should be replaced with wither Kotlin coroutines or `postStartupActivity`. I have used `postStartupActivity` for now. It runs when a project is opened inside the IDE it will run once every time a project is opened (similar to `ProjectOpenCloseListener`). It does not run when IDE is opened.

Reply:
Got it, thanks for explaining Saleem